### PR TITLE
Use the latest aws sdk gem

### PR DESCRIPTION
**Pull request by henrich-m**
Changes:
* Removed gem `aws-sdk` in favor of `aws-sdk-s3`.

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/310/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 8s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED

**Comment by henrich-m** [comment]
Hi guys, can you take a look in this @panjan @kubamahnert  ?
We need to update the ruby version on our main app but this gem is blocking it because off the Aws SDK fixed version. This PR will update the gem to the latest version and load only the related S3 files as the app only use it.

File: gooddata.gemspec [hunk 1]
```diff
@@ -49,7 +49,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'vcr'
 
   s.add_dependency 'activesupport', '~> 4.2.9'
-  s.add_dependency 'aws-sdk', '~> 2.9', '>= 2.9.42'
+  s.add_dependency 'aws-sdk-s3', '~> 1'
```
Comment (panjan) [inline gooddata.gemspec]: Please use the latest version when introducing dependencies `gem 'aws-sdk-s3', '~> 1.16'`.

Comment (panjan) [reply gooddata.gemspec]: We should update `Gemfile.lock` by running `bundle install`. This will ensure that a specific version is always used.

Comment (panjan) [reply gooddata.gemspec]: Can you please squash the two commits into one?

**Comment by panjan** [comment]
extended test

File: gooddata.gemspec (hunk 1 above)
Comment (henrich-m) [reply gooddata.gemspec]: @panjan I did the proper version locking in d4cec5d57, but I came to realize that the `Gemfile.lock` is ignored by the `.gitignore`, should I remove it from the ignore list and add to the repo anyway?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/315/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 7s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED

**Comment by henrich-m** [comment]
@panjan Just a friendly-reminder for checking this PR again :)

**Comment by panjan** [comment]
extended-test

File: gooddata.gemspec (hunk 1 above)
Comment (panjan) [reply gooddata.gemspec]: @henrich-m You're right. I've just realised that this repo is distributed as a Gem, therefore the Gemfile.lock should be ignored and only gemspec should be updated. Sorry for the confusion.

**Comment by panjan** [comment]
Tests failed with a [known error](https://jira.intgdc.com/browse/TMA-983). Merging.

**Comment by panjan** [comment]
@henrich-m Released in [1.2.0](https://rubygems.org/gems/gooddata/versions/1.2.0) 🚀

**Pull request merged**